corehq/apps/app_manager/xpath_validator/wrapper.py

```diff
@@ -17,7 +18,8 @@ def validate_xpath(xpath, allow_case_hashtags=False):
             cmd = ['node', path]
         p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
                              stderr=subprocess.PIPE)
-        stdout, stderr = p.communicate(xpath.replace('\r', ' ').encode('utf-8'))
+        # Collapse whitespace. '\r' mysteriously causes the process to hang in python 3.
+        stdout, stderr = p.communicate(re.sub(r'\s+', ' ', xpath).encode('utf-8'))
```

Comment:
@orangejenny 
whats the reason behind making `\r` to `r'\s+'`? last PR i thought it was just changing how whitespaces can appear on different systems but want to confirm my understanding now.

Reply:
My rationale was that if there are problems with `\r` there's an elevated chance there are problems with other whitespace sequences. Since xpath collapses whitespace, mimicking that behavior should be safe but also more generic than only stripping `\r`.

...that said, I wrote this PR when we had no idea what the problem was here. This is going to be superseded by https://github.com/dimagi/commcare-hq/pull/24875 I could just as well have closed the PR, but at least it adds in the comment.

Reply:
Thanks